app/controllers/concerns/paginates_answers.rb

```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module PaginatesAnswers
+  def paginate_answers
+    @answers = yield(last_id: params[:last_id])
+    answer_ids = @answers.map(&:id)
+    answer_ids += @pinned_answers.pluck(:id) if @pinned_answers.present?
+    @answers_last_id = answer_ids.min
+    @more_data_available = !yield(last_id: @answers_last_id, size: 1).count.zero?
+    Subscription.where(user: current_user, answer_id: answer_ids).pluck(:answer_id) if user_signed_in?
```

Comment:
this is a strange api (paginate_answers returns an id of subscriptions?)

instead, have this set an ivar instead of being a return

Reply:
e0bce1d